src/test/http-mocks/pay.ts

```diff
@@ -2,42 +2,66 @@ import * as mock from 'nock'
 import * as HttpStatus from 'http-status-codes'
 import * as config from 'config'
 
-const baseServiceURL: string = `${config.get('pay.url')}/users`
+const payUrl = config.get<string>('pay.url')
+const payPath = '/' + config.get<string>('pay.path')
 
-const payment = {
-  id: '12',
-  amount: 100,
-  state: {
-    status: 'failed'
-  },
+export const paymentInitiateResponse: object = {
+  reference: 'RC-1520-4225-4161-2265',
+  date_created: '2018-03-07T11:35:42.095+0000',
+  status: 'Initiated',
   _links: {
     next_url: {
-      method: 'get',
-      href: '/claim-confirmed'
+      href: 'https://www.payments.service.gov.uk/secure/8b647ade-02cc-4c85-938d-4db560404df8',
+      method: 'GET'
     }
   }
 }
 
+const paymentRetrieveResponse: object = {
+  amount: 60,
+  description: 'Money Claim issue fee',
+  reference: 'RC-1520-4276-0065-8715',
+  currency: 'GBP',
+  ccd_case_number: 'UNKNOWN',
+  case_reference: 'dfd75bac-6d54-4c7e-98f7-50e047d7c7f5',
+  channel: 'online',
+  method: 'card',
+  external_provider: 'gov pay',
+  status: 'Success',
+  external_reference: 'h8mtngl42o4i8ajrq64mdqufhl',
+  site_id: 'AA00',
+  service_name: 'Civil Money Claims',
+  fees: [
+    {
+      code: 'X0026',
+      version: '1',
+      calculated_amount: 60
+    }],
+  _links: {
+    self: { href: 'http://localhost:4421/card-payments/RC-1520-4276-0065-8715', method: 'GET' }
+  }
+}
+
 export function resolveCreate () {
-  mock(baseServiceURL)
-    .post(new RegExp('/[0-9]+/payment'))
-    .reply(HttpStatus.CREATED, { ...payment, state: { status: 'created' } })
+  mock(payUrl)
+    .post(payPath)
+    .reply(HttpStatus.CREATED, { ...paymentInitiateResponse })
```

Comment:
Please do `paymentInitiateResponse` instead of `{ ...paymentInitiateResponse }`. No need to unpack / spread source object here.

Reply:
If I don't unpack it doesn't deserialise into underlying type and test fails due to expose annotations we used for underscore property names.

Reply:
Nah, `paymentInitiateResponse` and `{ ...paymentInitiateResponse }` result in the same thing. I changed it to remove object spread.